App.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import {
+  View,
+  Text,
+  TouchableOpacity,
+  Platform,
+  StyleSheet
+} from 'react-native';
+import NfcManager, {NfcTech} from 'react-native-nfc-manager';
+import {Buffer} from 'buffer';
+
+interface State {
+  apduResponse: string;
+  testOutput: boolean;
+  detectedTechnology: string;
+  nfcUID: string;
+  showInstructions: boolean;
+}
+
+export class App extends React.Component<any, State> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    }
+  }
+
+  componentDidMount() {
+    NfcManager.start();
+  }
+
+  componentWillUnmount() {
+    this.cleanUp();
+  }
+
+  private cleanUp = () => {
+    NfcManager.cancelTechnologyRequest().catch(() => 0);
+    this.setState({
+      apduResponse: "Please run the test",
+      testOutput: false,
+      detectedTechnology: "",
+      nfcUID: "",
+      showInstructions: false
+    });
+  }
+
+  private test = async () => {
+    try {
+      let tech = NfcTech.IsoDep;
```

Comment:
prefer immutability
```suggestion
      const tech = NfcTech.IsoDep;
```
> Javascript Object Modification
 •  Due to its dynamic nature, JavaScript makes it extremely easy to modify objects that you do not own
 •  Such practice should generally be avoided as it can lead to maintenance problems and can produce hard to find bugs
 •  If you don't own it, don't modify it

Reply:
I agree... just the result of a badly checked cut-and-paste :-)